src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -2251,15 +2260,16 @@ def _save_importing_provenance(self):
             extension_group.attrs["info"] = info
 
     def _save_run_info(self):
-        run_info = self.run_info.copy()
-
-        if self.format == "binary_folder":
-            extension_folder = self._get_binary_extension_folder()
-            run_info_file = extension_folder / "run_info.json"
-            run_info_file.write_text(json.dumps(run_info, indent=4), encoding="utf8")
-        elif self.format == "zarr":
-            extension_group = self._get_zarr_extension_group(mode="r+")
-            extension_group.attrs["run_info"] = run_info
+        if self.run_info is not None:
+            run_info = self.run_info.copy()
+
+            if self.format == "binary_folder":
+                extension_folder = self._get_binary_extension_folder()
+                run_info_file = extension_folder / "run_info.json"
+                run_info_file.write_text(json.dumps(run_info, indent=4), encoding="utf8")
```

Comment:
One question I have here in general not related to this PR is if we encode with 'utf-8' will that then mess with other encoding systems? Or do we think that the encoding was really just limited to shell script issues? Is there any place where bouncing between encodings may bite us?

Reply:
Since we are building the dictionaries we are sure that they can be UTF-8 encoded and decoded, so I think it's safe!

Reply:
What about the case for paths where the paths include Chinese or Japanese characters, which can't be encoded with `uft-8`? For run_info that doesn't matter, but I'm wondering about other json files we make. It might be that we just let people know that this works for utf-8 and may work for other encodings. But since most people on the team all use utf-8 I think it will be hard for us to know for sure for this. Doesn't really hold anything up I guess.